src/main/resources/templates/index.html

```diff
@@ -42,7 +42,7 @@ <h3  aria-label="Recipient's username" aria-describedby="button-addon2" class="t
                                             <div class="input-group-prepend">
                                                 <span class="input-group-text">Amount $</span>
                                             </div>
-                                            <input class="form-control" id="amount" th:field="*{amount}" type="text" placeholder="0.00" aria-describedby="inputGroup-sizing-default">
+                                            <input class="form-control" id="amount" th:field="*{amount}" type="number" placeholder="0.00" aria-describedby="inputGroup-sizing-default">
```

Comment:
I changed the type of input from text to number because your income DTO variable name {amount} has a double object type. Here is the amount variable in your DTO: 

```java
private double amount;
```